mssqlcli/mssql_cli.py

```diff
@@ -353,9 +344,54 @@ def execute_command(self, text, query):
             elif query.meta_changed:
                 self.refresh_completions(persist_priorities='all')
 
-        return query
+        if not query.contains_secure_statement:
+            # Allow MssqlCompleter to learn user's preferred keywords, etc.
+            with self._completer_lock:
+                self.completer.extend_query_history(text)
+
+            self.query_history.append(query)
+
+        return output, query
+
+    def execute_query(self, text):
+        """ Processes a query string and outputs to file or terminal """
+
+        # Initialize default metaquery in case execution fails
+        query = MetaQuery(query=text, successful=False)
```

Comment:
We aren't doing anything in this method with the query. Can we move it's instantiation into the _execute_interactive_command method so that it doesn't have to be passed in via a parameter?